backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropCallConvertors.kt

```diff
@@ -123,10 +133,12 @@ private fun InteropCallContext.writeValueToMemory(
 
 private fun InteropCallContext.determineInMemoryType(type: IrType): IrType {
     val classifier = type.classOrNull!!
-    return if (classifier in symbols.unsignedIntegerClasses) {
-        symbols.unsignedToSignedOfSameBitWidth.getValue(classifier).owner.defaultType
-    } else {
-        type
+    return when {
+        // enum may have an unsigned base type so we perform a recursive call.
+        type.isCEnumType() -> determineInMemoryType(type.getCEnumPrimitiveType())
```

Comment:
Is this code reachable? Could you provide an example?

Reply:
It is reachable via `writeEnumValueToMemory`. I see that it is simpler to pass  `IrType.getCEnumPrimitiveType()` directly to `writeValueToMemory`.